TodoListClient/tests/acceptance/s/todo-items/index-test.js

```diff
@@ -0,0 +1,73 @@
+import { module, test } from 'qunit';
+import { setupApplicationTest } from 'ember-qunit';
+import { click, currentURL, findAll, visit } from '@ember/test-helpers';
+import { setupMirage } from 'ember-cli-mirage/test-support';
+import {
+  authenticateSession,
+  currentSession,
+  invalidateSession,
+} from 'ember-simple-auth/test-support';
+
+module('Acceptance | S | Todo Items | Index', (hooks) => {
+  setupApplicationTest(hooks);
+  setupMirage(hooks);
+
+  test('redirects to login if not authenticated', async function (assert) {
+    assert.expect(1);
+
+    await invalidateSession();
+    await visit('/s/todo-items');
+
+    assert.strictEqual(currentURL(), '/login', 'the route is correct');
+  });
+
+  test('inauthenticates the session when the user logs out', async function (assert) {
```

Comment:
adding a test for that would be tricky. we're handling our authentication w/ an addon called `ember-simple-auth`, and when in a testing environment, it will simply inauthenticate the session instead of triggering the full logout behavior, which includes redirecting the user and refreshing the page. we could override the addon's internals to get this to work, but i'd recommend against that since it would add to the complexity of our example app and could lead to future compatibility issues: https://github.com/mainmatter/ember-simple-auth/blob/master/packages/ember-simple-auth/addon/-internals/routing.js#L57

Reply:
Makes sense, let's leave it like this then.